dataprofiler/profilers/profile_builder.py

```diff
@@ -678,11 +693,11 @@ def tqdm(l):
         # Create structured profile objects
         new_cols = set()
         for col in df.columns:
-            if col not in profile:
+            if col not in self._profile:
```

Comment:
I don't think any of these `self` are necessary. Also, I don't think you'd need to return profile if you were just doing it this way.

Reply:
Could you explain further? this code no longer returns self._profile